src/main/java/com/cloudcomputing/cloudkarma/Actions.java

```diff
@@ -22,142 +21,177 @@
+		return describeTasksRunningInCluster(cluster).getTaskArns().size() == 0;
+	}
+
+	private static List<String> getAutoScalingGroupNameFromInstanceId(List<String> instanceIds) {
+		Set<String> autoScalingGroupNameSet = new HashSet<>();
+		AmazonAutoScalingClient autoScalingClient = getAutoScalingClient(Regions.US_WEST_2);
+		DescribeAutoScalingInstancesRequest request = new DescribeAutoScalingInstancesRequest();
+		request.setInstanceIds(instanceIds);
+		DescribeAutoScalingInstancesResult result = autoScalingClient.describeAutoScalingInstances(request);
+		result.getAutoScalingInstances().forEach(
+				autoScalingInstance -> autoScalingGroupNameSet.add(autoScalingInstance.getAutoScalingGroupName()));
+		return autoScalingGroupNameSet.stream().collect(Collectors.toList());
+	}
+
+	private static SetDesiredCapacityResult updateAutoScaleGroup(String autoScalingGroupName, int capacity) {
+		AmazonAutoScalingClient autoScalingClient = getAutoScalingClient(Regions.US_WEST_2);
+		SetDesiredCapacityRequest request = new SetDesiredCapacityRequest();
+		request.setAutoScalingGroupName(autoScalingGroupName);
+		request.setDesiredCapacity(capacity);
+		SetDesiredCapacityResult result = autoScalingClient.setDesiredCapacity(request);
+		return result;
+	}
+
+	private static ListTasksResult describeTasksRunningInCluster(String cluster) {
+		AmazonECSClient ecsClient = getEcsClient(Regions.US_WEST_2);
+		ListTasksRequest request = new ListTasksRequest();
+		request.setCluster(cluster);
+		ListTasksResult result = ecsClient.listTasks(request);
+		return result;
+	}
+
+	private static ListContainerInstancesResult getInstancesForTheCluster(String cluster) {
+		AmazonECSClient ecsClient = getEcsClient(Regions.US_WEST_2);
+		ListContainerInstancesRequest request = new ListContainerInstancesRequest();
+		request.setCluster(cluster);
+		request.setStatus(ContainerInstanceStatus.ACTIVE);
+		ListContainerInstancesResult result = ecsClient.listContainerInstances(request);
+		return result;
+	}
+
+	private static List<String> getContainerInstanceIds(List<String> inputContainerInstanceIds) {
+		// extracts the ec2instance-id from the container instance ARN
+		List<String> containerInstanceIdList = new ArrayList<>();
+		Pattern p = Pattern.compile("(.+\\/)(.+)");
+		inputContainerInstanceIds.forEach(containerInstanceId -> {
+			Matcher matcher = p.matcher(containerInstanceId);
+			if (matcher.find())
+				containerInstanceIdList.add(matcher.group(2));
+		});
+		return containerInstanceIdList;
+	}
+
+	private static List<RemainingResources> getContainerResources(List<String> containerInstances) {
+		DescribeContainerInstancesResult result = describeContainerInstances(containerInstances);
+
+		List<RemainingResources> resources = new ArrayList<>();
+
+		result.getContainerInstances().forEach(containerInstance -> {
+			RemainingResources remainingResources = new RemainingResources(containerInstance.getEc2InstanceId(),
```

Comment:
Use lombok's constructor.

Reply:
Did not use constructor. See above comments. Please suggest code to explain.

Reply:
Use getter to set the values 
or 
use @AllArgsConstructor and pass these values to constructor